app/src/main/java/com/android/solvit/seeker/ui/provider/ProviderRegistrationProfile.kt

```diff
@@ -325,3 +460,335 @@ fun ProviderRegistrationScreen(
             }
       })
 }
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ProviderDetails(
+    selectedService: String,
+    onSelectedServiceChange: (String) -> Unit,
+    description: String,
+    onDescriptionChange: (String) -> Unit,
+    startingPrice: String,
+    onStartingPriceChange: (String) -> Unit,
+    selectedLanguages: MutableList<String>,
+    providerImageUri: Uri?,
+    onImageSelected: (Uri?) -> Unit
+) {
+
+  var servicesExpanded by remember { mutableStateOf(false) }
+  var languagesExpanded by remember { mutableStateOf(false) }
+
+  val services = Services.entries.toTypedArray()
+  val availableLanguages = Language.entries.toList().map { it.toString() }
+
+  val isDescriptionOk = description.isNotBlank()
+  val isStartingPriceOk = startingPrice.isNotBlank() && startingPrice.all { it.isDigit() }
+
+  Column(
+      modifier = Modifier.fillMaxSize().padding(16.dp),
+      verticalArrangement = Arrangement.spacedBy(16.dp)) {
+        // DropDown Menu to select provider's service
+        ExposedDropdownMenuBox(
+            expanded = servicesExpanded,
+            onExpandedChange = { servicesExpanded = !servicesExpanded },
+            modifier = Modifier.testTag("servicesDropDown")) {
+              OutlinedTextField(
+                  value = selectedService,
+                  onValueChange = { onSelectedServiceChange(it) },
+                  label = { Text("What Services Do You Offer?") },
+                  readOnly = true,
+                  modifier = Modifier.fillMaxWidth().menuAnchor())
+              ExposedDropdownMenu(
+                  expanded = servicesExpanded, onDismissRequest = { servicesExpanded = false }) {
+                    services.forEach { service ->
+                      DropdownMenuItem(
+                          modifier = Modifier.testTag("$service"),
+                          text = { Text(text = service.toString()) },
+                          onClick = {
+                            onSelectedServiceChange(service.toString())
+                            servicesExpanded = false
+                          })
+                    }
+                  }
+            }
+
+        // Upload photo provider section
+        UploadImage(
+            selectedImageUri = providerImageUri,
+            imageUrl = null,
+            onImageSelected = { uri -> onImageSelected(uri) })
+
+        // Enter Provider Brief description
+        CustomOutlinedTextField(
+            value = description,
+            onValueChange = { onDescriptionChange(it) },
+            label = "About you",
+            placeholder =
```

Comment:
There is no limit for the description size. I don't know if it should be taken care of here or where the description is used but I was just thinking about it.

Reply:
The rest LGTM!

Reply:
Yes ! we can add a condition ! What about a max of 50 words ?